File: app/src/main/java/me/fru1t/android/annotations/VisibleForXML.java
package me.fru1t.android.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Denotes that a method is public for XML use.
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.SOURCE)
public @interface VisibleForXML { }

File: app/src/main/java/me/fru1t/android/annotations/VisibleForXML.kt
package me.fru1t.android.annotations

/** Denotes that a method is public for XML use. */
@Target(
        AnnotationTarget.FUNCTION,
        AnnotationTarget.PROPERTY_GETTER,
        AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.SOURCE)
annotation class VisibleForXML
